src/attr.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt;
+use std::borrow::Cow;
+
+use escape::Escaped;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Attribute<'a> {
+    pub name: Cow<'a, str>,
+    pub value: Cow<'a, str>,
+}
+
+#[derive(Clone, Debug)]
+pub struct AttributeSet<'a> {
+    pub attrs: Cow<'a, [Attribute<'a>]>,
+}
+
+impl<'a> AttributeSet<'a> {
+    pub fn empty() -> AttributeSet<'a> {
+        AttributeSet::new(Cow::Borrowed(&[]))
+    }
+
+    pub fn new(attrs: Cow<'a, [Attribute<'a>]>) -> AttributeSet<'a> {
+        AttributeSet { attrs: attrs }
+    }
+}
+
+impl<'a, T> From<T> for AttributeSet<'a>
+    where T: Into<Cow<'a, [Attribute<'a>]>>
+{
+    fn from(attrs: T) -> AttributeSet<'a> {
+        AttributeSet::new(attrs.into())
+    }
+}
```

Comment:
Having multiple implementations of From with generics is currently impossible (maybe just really difficult) because of the coherence rules. Could either provide implementations for all the potential types in std we want to support, or I guess write out all the potential cases and see what different generic impls cover.